cmd/helm/env.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"helm.sh/helm/cmd/helm/require"
+)
+
+var envHelp = `
+Env prints out all the environment variables in use by Helm.
+`
+
+func newEnvCmd(out io.Writer) *cobra.Command {
+	o := &envOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "env",
+		Short: "environment variables used by the Helm client",
+		Long:  envHelp,
+		Args:  require.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return o.run(out)
+		},
+	}
+
+	return cmd
+}
+
+type envOptions struct {
+	maxEnvKeyLength int
+}
+
+// run .
+func (o *envOptions) run(out io.Writer) error {
+
+	helmEnvs := make(map[string]string)
+	xdgEnvs := make(map[string]string)
+	o.maxEnvKeyLength = 0
+
+	for _, e := range os.Environ() {
```

Comment:
Try to think of this like `go env`. It prints environment information including empty environment variables and defaults. It can't just look at the current environment variables. For example, in Go if I don't have a `GOPATH` set it defaults to `~/go` and that shows up if `go env` is run.

Can you please update it work in a similar manner.